### PR TITLE
Update interfaceToC.dd

### PR DESCRIPTION
**Pull request by Dashster**
The string literal passed to printf appears to have a null character appended. The documentation currently says that it doesn't, and that it does. Change first sentence to say that it does.

### CODE REVIEW

File: spec/interfaceToC.dd [hunk 1]
```diff
@@ -227,7 +227,7 @@ void foo(char[] string)
 ------
 
         $(P The $(CODE printf) format string literal
-        in the example doesn't end with $(CODE '\0').
+        in the example DOES end with $(CODE '\0').
```
Comment (wilzbach) [inline spec/interfaceToC.dd]: Full capitalization isn't used in spec. How about `$(I ...)` for italic?

Comment (Dashster) [reply spec/interfaceToC.dd]: Changed to $(I does).

**Comment by aG0aep6G** [comment]
As far as I see, this change is wrong.

The string literal in the example does **not** end with `\0`. A string literal that ends with `\0` would be: `"foo\0"`. As the documentation says, the null character is stored **after** the end of the literal. The documentation is correct as it is. But maybe it could be improved to convey the message more clearly.

**Comment by wilzbach** [comment]
@Dashster have a look at https://en.wikipedia.org/wiki/Printf_format_string#Precision_field

> For the string type, it limits the number of characters that should be output, after which the string is truncated.

tl;dr:  "printf will still see one" is incorrect, it's just a specific printf instruction.

**Comment by aG0aep6G** [comment]
> @Dashster have a look at https://en.wikipedia.org/wiki/Printf_format_string#Precision_field
[...]
> tl;dr: "printf will still see one" is incorrect, it's just a specific printf instruction.

I think you're misunderstanding this. The text is not about the `string` variable (which should probably have a different name). It's about the format string `"my string is: %.*s\n"`. The `%.*s` in there doesn't matter. The code could just as well be `printf("foo");`.

**Comment by Dashster** [comment]
I would like to withdraw this request but don't know how to do so. If anyone else can do so, I would appreciate it.
